Uses qs module to parse query-string

echetzakis opened the pull request:
Koa's internal query parser does not deserialize deep objects
from the query-string, hence koa-oas3 does not enforce
the schema validation for these parameters.

Using qs module to parse the query-string before validation
addresses this limitation.

echetzakis commented:
Hi @supertong, 

Since I'm not familiar with your `Changeset` tool, I don't understand why it fails.
If you find this PR interesting let me know what else is needed in order to pass this check.

Kind Regards,
Eleftherios

supertong commented on package.json:
```diff
@@ -25,12 +25,14 @@
     ]
   },
   "dependencies": {
+    "@types/qs": "^6.5.3",
```
Can we move this to `devDependency`?

supertong commented:
Hey @echetzakis, thanks for contribution. LGTM. Would you be able to resolve the conflict?
Don't worry about the changeset check for now, I think I didn't set it up properly.

echetzakis commented:
Hi @supertong,
Rebased on your latest master and moved `@types/qs` to dev dependencies.

Thanks for considering my contribution. 
I love this project (and chow-chow) and use it in all my projects!
👍
